tests/test_util.py

```diff
@@ -54,6 +61,43 @@ def test_with_statement(self):
         self.assertEqual(l, [1, 2, 3, 4])
 
 
+class TestProblemCache(unittest.TestCase):
+    def setUp(self):
+        self.reference = TESTMODEL.solve().fluxes
+        self.n_constraints = len(TESTMODEL.solver.constraints)
+        self.n_variables = len(TESTMODEL.solver.variables)
+
+    def test_cache_problem(self):
+        lmoma(model=TESTMODEL, reference=self.reference)
+
+        # After running lmoma without cache, the number of variables and constraints remains the same
+        self.assertEqual(self.n_constraints, len(TESTMODEL.solver.constraints))
+        self.assertEqual(self.n_variables, len(TESTMODEL.solver.variables))
+
+        cache = ProblemCache(TESTMODEL)
+        lmoma(model=TESTMODEL, reference=self.reference, cache=cache)
+        # After running lmoma with cache, the number of variables is 2 times the original number
+        self.assertEqual(2*self.n_variables, len(TESTMODEL.solver.variables))
+        # And has 2 more constraints per reactions
+        self.assertEqual(self.n_constraints + 2 * len(TESTMODEL.reactions), len(TESTMODEL.solver.constraints))
+
+        cache.reset()
+        # After reset cache, the problem should return to its original size
+        self.assertEqual(self.n_constraints, len(TESTMODEL.solver.constraints))
+        self.assertEqual(self.n_variables, len(TESTMODEL.solver.variables))
+
+    def test_with(self):
+        with ProblemCache(TESTMODEL) as cache:
+            lmoma(model=TESTMODEL, reference=self.reference, cache=cache)
+            # After running lmoma with cache, the number of variables and constraints are 2 times bigger
+            self.assertEqual(2*self.n_variables, len(TESTMODEL.solver.variables))
+            # And has 2 more constraints per reactions
+            self.assertEqual(self.n_constraints + 2 * len(TESTMODEL.reactions), len(TESTMODEL.solver.constraints))
+        # After reset cache, the problem should return to its original size
+        self.assertEqual(self.n_constraints, len(TESTMODEL.solver.constraints))
+        self.assertEqual(self.n_variables, len(TESTMODEL.solver.variables))
+
```

Comment:
It would be nice with some unittests that test the API of ProblemCache, i.e. a test for add_constraint, add_variable, etc.